src/Superpower/Combinators.cs

```diff
@@ -1034,8 +1034,9 @@ public static TextParser<T> Try<T>(this TextParser<T> parser)
         /// <typeparam name="T">The type of value being parsed.</typeparam>
         /// <param name="parser">The parser.</param>
         /// <param name="predicate">The predicate to apply.</param>
+        /// <param name="message">An optional error message when parsing fails.</param>
         /// <returns>The resulting parser.</returns>
-        public static TokenListParser<TKind, T> Where<TKind, T>(this TokenListParser<TKind, T> parser, Func<T, bool> predicate)
+        public static TokenListParser<TKind, T> Where<TKind, T>(this TokenListParser<TKind, T> parser, Func<T, bool> predicate, string message = "unsatisfied condition")
```

Comment:
If `message` is nullable then the type should be `string?`. Or if not, then an argument check is still required to catch misuse by consumers on earlier language versions. (Making it `string?` seems reasonable to me, as it would match `Result<T>.ErrorMessage`.)

Reply:
@nblumhardt  In which cases does it make sense to have the message to be `null`?

Reply:
🤔 - probably aren't any. `string` and the null check would be fine 👍

Reply:
@nblumhardt added.